compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -253,12 +264,20 @@ fn print_target_features(sess: &Session, tm: &llvm::TargetMachine) {
     let mut rustc_target_features = supported_target_features(sess)
         .iter()
         .filter_map(|(feature, _gate)| {
-            let llvm_feature = to_llvm_feature(sess, *feature);
-            // LLVM asserts that these are sorted. LLVM and Rust both use byte comparison for these strings.
-            target_features.binary_search_by_key(&llvm_feature, |(f, _d)| *f).ok().map(|index| {
-                let (_f, desc) = target_features.remove(index);
-                (*feature, desc)
-            })
+            for llvm_feature in to_llvm_feature(sess, *feature) {
+                // LLVM asserts that these are sorted. LLVM and Rust both use byte comparison for these strings.
+                match target_features
+                    .binary_search_by_key(&llvm_feature, |(f, _d)| (*f).to_string())
```

Comment:
```suggestion
                    .binary_search_by_key(&llvm_feature, |(f, _d)| f)
```

should work.

Reply:
This dissolved away when I switched back to &str above.